### PR TITLE
Fix multiple cases of invalid windows line endings handling

### PR DESCRIPTION
**Pull request by lukaszsamson**

### CODE REVIEW

File: apps/language_server/lib/language_server/providers/formatting.ex [hunk 1]
```diff
@@ -71,7 +71,7 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
 
   defp advance_pos({line, col}, str) do
     Enum.reduce(String.split(str, "", trim: true), {line, col}, fn char, {line, col} ->
-      if char == "\n" do
+      if char in ["\n", '\r'] do
```
Comment (axelson) [inline apps/language_server/lib/language_server/providers/formatting.ex:74]: Shouldn't this be `"\r"`?

```suggestion
      if char in ["\n", "\r"] do
```

File: apps/language_server/lib/language_server/diagnostics.ex [hunk 2]
```diff
@@ -21,7 +21,7 @@ defmodule ElixirLS.LanguageServer.Diagnostics do
     {reversed_stacktrace, reversed_description} =
       diagnostic_message
       |> String.trim_trailing()
-      |> String.split("\n")
+      |> String.split(["\r\n", "\r", "\n"])
```
Comment (axelson) [inline apps/language_server/lib/language_server/diagnostics.ex:24]:
```suggestion
      |> SourceFile.lines()
```

File: apps/language_server/lib/language_server/providers/execute_command.ex [hunk 3]
```diff
@@ -55,7 +55,7 @@ defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand do
 
         Code.format_string!("@spec #{spec}", line_length: target_line_length)
         |> IO.iodata_to_binary()
-        |> String.split("\n")
+        |> String.split(["\r\n", "\r", "\n"])
```
Comment (axelson) [inline apps/language_server/lib/language_server/providers/execute_command.ex:58]:
```suggestion
        |> SourceFile.lines()
```

**Comment by lukaszsamson** [comment]
> This is a good change, but for consistency I think we should use `SourceFile.lines/1` wherever we can.

I wasn't sure if it wouldn't be misleading as those are not source files

**Pull request merged**